src/Utility/Log.cpp

```diff
@@ -1,59 +1,148 @@
 #include "Log.hpp"
 
-#include <cstdio>
+#include <ostream>
+#include <iostream>
 
 using namespace std;
 
+ostream* Log::streams[NUMBER_OF_CHANNELS];
+
+Log::Log(const Channel channel) {
+    currentChannel = channel;
+}
+
 Log::~Log() {
     fflush(stderr);
 }
 
-Log& Log::operator<<(string text) {
-    fputs(text.c_str(), stderr);
+Log& Log::operator<<(const string& text) {
+    *streams[currentChannel] << text;
+
+    #ifdef LOGTESTING 
+    if (currentChannel == ERR)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') keyword 'if' not followed by a single space [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)